### DISCUSSION TITLE
Make Integrations chainable

### DISCUSSION THREAD

**Discussion by rmeissner** [5afe/safe-core-protocol-specs#14] [Ideas]
It should be possible to compose/ chain multiple plugins/hooks. This would allow for a better composability, i.e. it would be possible to reuse a time lock with different other plugins. 

Open Questions:
- Should guards be applied to each component in the chain (or only last)?
- Should registry checks be performed for each component in the chain (or only last)?

**Comment by auryn-macmillan** [comment]
In [Zodiac](https://github.com/gnosis/zodiac), we have a class of Modules called "Modifiers".
Modifiers expose `IAvatar` (the `ModuleManager.sol` interface) so that any modules built to call `execTransactionFromModule()` / `execTransactionFromModuleReturnData()` can target the Modifier, rather than the safe that will ultimately execute the call, in order to create a chain of mods.

[Delay](https://github.com/gnosis/zodiac-modifier-delay) and [Roles](https://github.com/gnosis/zodiac-modifier-roles) are the two canonical examples of Modifiers in Zodiac.

In Safe Protocol, we could achieve a similar outcome by having `Modifiers` expose `ISafeProtocolMediator`, so that other mods can target modifiers, rather than the mediator.
